Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/TemplatedProjectExporter.cs

```diff
@@ -218,26 +224,350 @@ public void GenerateDirectoryPropsFile(UnityProjectInfo unityProjectInfo)
+            dependencyProjectMapping.AddConfigurationMapping(new ConfigPlatformPair("Release", "Any CPU"), "ActiveCfg", "Build.0");
+
+            List<string> anyCpuPlatform = new List<string> { "Any CPU" };
+            ProcessDefaultMappings(dependencyProjectMapping, new Dictionary<string, List<string>> { { "Debug", anyCpuPlatform }, { "Release", anyCpuPlatform } });
+
+            // Process projects that we aren't generating, but were added to the solution file
+            if (solutionFileInfo != null)
+            {
+                foreach (KeyValuePair<Guid, List<ProjectConfigurationEntry>> pair in solutionFileInfo.ProjectConfigurationEntires)
+                {
+                    if (solutionFileInfo.Projects.ContainsKey(pair.Key))
+                    {
+                        projectOrdering.Add(pair.Key);
+
+                        ProjectConfigurationMapping mapping = new ProjectConfigurationMapping();
+                        projectConfigMapping.Add(pair.Key, mapping);
+
+                        foreach (ProjectConfigurationEntry item in pair.Value)
+                        {
+                            mapping.Mappings[item.SolutionConfig] = item.ProjectConfig;
+                            mapping.GetPropertySet(item.SolutionConfig).Add(item.Property);
+                        }
+                    }
+                }
+            }
+
+            ProcessMappings(configPlatformPropertyTemplate, rootReplacementSet, projectOrdering, projectConfigMapping);
+        }
+
+        private void ProcessPropertiesSet(ITemplatePart solutionPropertiesTemplate, TemplateReplacementSet rootReplacementSet, Dictionary<string, string> propertySet, Dictionary<string, string> existingPropertySet)
+        {
+            void WriteSolutionProperty(string key, string value)
+            {
+                TemplateReplacementSet replacementSet = solutionPropertiesTemplate.CreateReplacementSet(rootReplacementSet);
+                solutionPropertiesTemplate.Tokens["PROPERTY_KEY"].AssignValue(replacementSet, key);
+                solutionPropertiesTemplate.Tokens["PROPERTY_VALUE"].AssignValue(replacementSet, value);
+            }
+
+            foreach (KeyValuePair<string, string> prop in propertySet)
+            {
+                WriteSolutionProperty(prop.Key, prop.Value);
+            }
+
+            if (existingPropertySet != null)
+            {
+                foreach (KeyValuePair<string, string> prop in existingPropertySet)
+                {
+                    if (!propertySet.ContainsKey(prop.Key))
+                    {
+                        WriteSolutionProperty(prop.Key, prop.Value);
+                    }
+                }
+            }
+        }
+
+        private SortedDictionary<string, SortedSet<string>> GetSolutuonConfigurationPlatformMap(UnityProjectInfo unityProjectInfo, SolutionFileInfo solutionFileInfo)
```

Comment:
>GetSolutuonConfigurationPlatformMap [](start = 60, length = 35)

nit: "GetSolutuonConfigurationPlatformMap" should be "GetSolutionConfigurationPlatformMap"

Reply:
Good catch.

---
In reply to: [364495690](https://github.com/microsoft/MSBuildForUnity/pull/96#discussion_r364495690) [](ancestors = 364495690)